src/main/java/com/google/javascript/cl2dts/DeclarationGenerator.java

```diff
@@ -166,11 +168,20 @@ private void declareNamespace(String namespace, String provide, TypedVar symbol,
       new TreeWalker(symbol, namespace).walk(true);
     } else {
       // JSCompiler treats "foo.x" as one variable name, so collect all provides that start with
-      // $provide + ".".
-      String prefix = provide + ".";
+      // $provide + "." but are not sub-properties.
+      Set<String> desiredSymbols = new HashSet<>();
+
+      TypedVar o = Iterables.find(allSymbols, nameIs(provide));
```

Comment:
Yes, good observation. I had something else here and refactored to this... earlier I had the top level symbol instead of the one for this provide